### PR TITLE
Proposal for a toggle setting for mouseover marker 

### PR DESCRIPTION
**Pull request by bmorris3**
### Description

After sprint planning today, @kecnry and I discussed some ideas for how to toggle the mouseover marker and coordinate info. We talked about some of the difficulties with adding a new button to the toolbar, and decided we might scope out alternatives first. 

This PR is a small addition to #1894 which toggles the mouseover marker and coordinate info with a switch in the Plot Options plugin, under the Settings drop-down. Demo below:

https://user-images.githubusercontent.com/3497584/213061391-b3a17923-39d0-463b-9827-156c4f51dcee.mov

I decided to contribute this suggestion as a PR to @pllim's PR, since (1) it depends on @pllim's implementation in #1894, and (2) it has a pretty small diff. Happy to make a PR to base:spacetelescope/jdaviz if that's better.

[🐱](https://jira.stsci.edu/browse/JDAT-3038)

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/configs/specviz/plugins/viewers.py [hunk 1]
```diff
@@ -65,6 +65,11 @@ def on_mouse_or_key_event(self, data):
             else:  # pragma: no cover
                 return
 
+        # only handle mouseover markers if the setting is selected:
+        plot_options_plugin = self.jdaviz_app.get_tray_item_from_name('g-plot-options')
+        if not plot_options_plugin.show_mouseover_marker:
+            return
```
Comment (kecnry) [inline jdaviz/configs/specviz/plugins/viewers.py:71]: I think we want only the marker to be toggled, but still show the information in the panel.  So `self.label_mouseover.marks[self._reference_id].visible = plot_options_plugin.show_mouseover_marker` somewhere.

File: jdaviz/configs/default/plugins/plot_options/plot_options.vue [hunk 2]
```diff
@@ -19,6 +19,14 @@
                 persistent-hint
               ></v-switch>
             </v-row>
+            <v-row>
+              <v-switch v-if="config=='specviz'"
```
Comment (kecnry) [inline jdaviz/configs/default/plugins/plot_options/plot_options.vue:23]: cubeviz and mosviz also have spectral viewers, so I think we want the switch there as well.

```suggestion
              <v-switch v-if="['specviz', 'cubeviz', 'mosviz'].indexOf(config) !== -1"

```

**Review by kecnry — COMMENTED** [review]
I am beginning to like this place for the switch more and more (and might even prefer it to eventually moving it to a "markers" plugin).  Thanks!

**Comment by kecnry** [comment]
If we move forward in this direction, I think we'll probably also want to consider removing the `locking_active` logic (to always act as `True`) to make this so that the switch alone determines whether the marker is always or never shown.

File: jdaviz/configs/specviz/plugins/viewers.py (hunk 1 above)
Comment (bmorris3) [reply jdaviz/configs/specviz/plugins/viewers.py:71]: I think the only place where we assign the LHS of the above snippet to True is here: 

https://github.com/pllim/jdaviz/blob/da4a704e29861ca6f1553d1cbf663b4601905fd6/jdaviz/configs/specviz/plugins/viewers.py#L162

so I'll change that `True` to `plot_options_plugin.show_mouseover_marker`

Comment (kecnry) [reply jdaviz/configs/specviz/plugins/viewers.py:71]: right.  Alternatively, the switch could act as the current `locking_active` (but probably renamed), and the text in the plugin changed to reflect that.  That actually would probably be my vote (the switch controls both the marker AND exposing the coordinates of the nearest spectrum, but the mouse position is always exposed).

**Comment by pllim** [comment]
https://github.com/spacetelescope/jdaviz/pull/1894 is merged, so I am closing this here. Feel free to resubmit this patch as PR straight to Jdaviz. Thanks!

**Pull request closed without merging**